### PR TITLE
check for missing item list returned due to k8b bug[1]

### PR DESCRIPTION
**Pull request by moolitayer**
[1] https://github.com/kubernetes/kubernetes/issues/13096

### CODE REVIEW

**Comment by moolitayer** [comment]
@abonas @simon3z @zakiva

File: lib/kubeclient/common.rb [hunk 1]
```diff
@@ -172,7 +172,9 @@ def get_entities(entity_type, klass, options)
             result.fetch('metadata', {}).fetch('resourceVersion', nil)
       end
 
-      collection = result['items'].map { |item| new_entity(item, klass) }
+      # result['items'] might be nil due to https://github.com/kubernetes/kubernetes/issues/13096
+      collection = []
+      collection = result['items'].map { |item| new_entity(item, klass) } if result['items']
```
Comment (simon3z) [inline lib/kubeclient/common.rb]: @moolitayer I think that just:

    collection = result['items'].to_a.map { |item| new_entity(item, klass) } if result['items']

would do the trick.

**Comment by simon3z** [comment]
@moolitayer 1 style comment, rest is good

File: lib/kubeclient/common.rb (hunk 1 above)
Comment (simon3z) [inline lib/kubeclient/common.rb]: Sorry I meant:

    collection = result['items'].to_a.map { |item| new_entity(item, klass) }

**Comment by simon3z** [comment]
LGTM :+1: ready for merge cc @abonas

File: lib/kubeclient/common.rb [hunk 2]
```diff
@@ -172,7 +172,8 @@ def get_entities(entity_type, klass, options)
             result.fetch('metadata', {}).fetch('resourceVersion', nil)
       end
 
-      collection = result['items'].map { |item| new_entity(item, klass) }
+      # result['items'] might be nil due to https://github.com/kubernetes/kubernetes/issues/13096
+      collection = result['items'].to_a.map { |item| new_entity(item, klass) } if result['items']
```
Comment (simon3z) [inline lib/kubeclient/common.rb]: @moolitayer you forgot to remove `if result['items']`

Comment (moolitayer) [inline lib/kubeclient/common.rb]: 10x

**Comment by abonas** [comment]
LGTM

**Pull request merged**